index.rst

```diff
@@ -326,33 +326,167 @@ Analysis of unbinned data is clearly needed for pipeline data quality analyses,
+
+The current plan is to have a Butler repository per instrument at the Summit.
+
+Once the exposures are ingested into the Butler "raw" collection, IDs are assigned, and the registry becomes the primary source for obtaining the list of observations.
+
+The Butler registry schema is described in `DMTN-073`_.
+It is designed to support visits composed of multiple exposures.
+If the visit is composed of a single exposure, then the visit table has entries that are essentially duplicates of the science exposures in the exposure table.
+
+However, the registry database contains only a small subset of the raw image metadata (the information about the conditions under which an image was taken summarized in the image headers).
+
+Analysis outputs are not in the registry database; they are typically linked to the visit table in the registry and stored as files in the Butler datastore.
+An example of an SFP task that produces such ouput is the `ConsolidateVisitSummaryTask`_ task that persists per-detector summary statistics computed for calibrated exposures into a per-visit dataset in Parquet files.
+
+Expanding the Butler registry database to include the required raw image metadata and analysis outputs is outside the plans.
+Hence the need for a :ref:`Visit database <visit-database>` (see also `DMTN-227 The Consolidated Database of Image Metadata`_).
+
+.. _visit-database:
+
+The Visit database
+""""""""""""""""""
+
+The initial content for the Visit database is described in the database `use cases`_ confluence page and summarized in this section.
+We also discuss implementation details for creating and updating the Visit database tables.
+
+Similarly to the Butler registry, a PostgreSQL database is the natural choice for the Visit database implementation.
+The Visit database should be replicated at the USDF, and the Summit instance should keep data for at least 30 days.
+
+**Visit and detector tables**
+
+The image headers and the Rapid Analysis outputs are the primary sources of information for the visit and detector tables.
+
+Rapid Analysis will have access to the headers during processing and can consolidate the raw image metadata into per-detector, and per-visit datasets along with the per-visit summary statistics from SFP persisted as Butler datasets.
+Scalar metrics (see :ref:`analysis tools overview <analysis_tools_overview>`) can also be persisted as Butler datasets (see `DMTN-203 Tracking Metrics in Butler`_).
+
+The mechanism for creating these tables in a relational database already exists in DM.
+In particular, if the schema for a Butler dataset is defined in the `Science Data Model`_ (SDM) repository, those definitions can be used to create the corresponding tables.
+
+For the Rapid analysis use case, however, a different mechanism than the one used for data release processing to update the database tables is required, given that the information must be readily available after each visit is processed.
```

Comment:
Why?  Is there an intrinsic delay in the pre-existing method?

Reply:
Yes, the performance of querying datasets in the registry, open the files from the butler data store and load the data into the database is known to be slow and it is ok if you have time to load the data after the processing. That's the mechanism we use for DRP. What I'm trying to say is that  this mechanism won't work for RA. For RA we should write the dataset records directly to the database. Another way to think about this is "extract and load" vs. streaming data.

Reply:
I suggest just adding a sentence just saying why the current DRP method is slow and why it may not work.